src/main/java/core/basesyntax/service/strategy/OperationStrategyImpl.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.service.strategy;
+
+import core.basesyntax.model.OperationType;
+import core.basesyntax.model.impl.FruitTransaction;
+import core.basesyntax.service.OperationHandler;
+import core.basesyntax.service.OperationStrategy;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy<FruitTransaction> {
+    private final Map<OperationType, OperationHandler<FruitTransaction>> operationHandlerMap;
+
+    public OperationStrategyImpl(Map<OperationType,
+            OperationHandler<FruitTransaction>> operationHandlerMap) {
+        this.operationHandlerMap = operationHandlerMap;
+    }
+
+    @Override
+    public OperationHandler<FruitTransaction> get(FruitTransaction transaction) {
+        return operationHandlerMap.get(transaction.getTransactionType());
+    }

```

Comment:
this whole class is essentially for just incapsulating the map in here, but I think that is complicating things, so I would just use this map directly in the `FruitTransactionProcesor`